protoplugin_test.go

```diff
@@ -68,6 +69,41 @@ func TestBasic(t *testing.T) {
 	)
 }
 
+func TestWithVersionOption(t *testing.T) {
+	t.Parallel()
+
+	run := func(args []string, runOptions ...RunOption) (string, error) {
+		stdout := bytes.NewBuffer(nil)
+		err := Run(
+			context.Background(),
+			args,
+			iotest.ErrReader(io.EOF),
+			stdout,
+			io.Discard,
+			HandlerFunc(func(ctx context.Context, w *ResponseWriter, r *Request) error { return nil }),
+			runOptions...,
+		)
+		return stdout.String(), err
+	}
+
+	_, err := run([]string{"--unsupported"})
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "unknown argument")
```

Comment:
We should make a private unknownArgumentsError that takes a string slice, prints out the conditional plural depending on length of args, and do errors.Is here, just to get in habit of not testing error strings in this package while it's still greenfield

Reply:
Sure, updated in 9f4f65cf.